tests/ui/traits/non_lifetime_binders/unifying-placeholders-in-query-response-2.rs

```diff
@@ -0,0 +1,23 @@
+// revisions: current next
+//[next] compile-flags: -Znext-solver
+// check-pass
+
+#![feature(non_lifetime_binders)]
+//~^ WARN the feature `non_lifetime_binders` is incomplete
+
+trait Id {
+    type Output: ?Sized;
+}
+
+impl<T: ?Sized> Id for T {
+    type Output = T;
+}
+
+trait Everyone {}
+impl<T: ?Sized> Everyone for T {}
+
+fn hello() where for<T> <T as Id>::Output: Everyone {}
```

Comment:
I'm somewhat concerned that this doesn't fail if we, for example, replace with this `for<'a> <&'a () as Id>::Output`, which *should* have this same issue? 🤔

Reply:
I think this is also an issue for regions, but we may never canonicalize placeholders from a non-root universe?

in MIR typeck we instantiate the binder in the query, so it doesn't contain placeholders.

Method probing also only has root variables :thinking: 

Where else do we use the old canonicalizer?